### PR TITLE
Fix json files cleanup when generating manifests

### LINKED ISSUE #498: Bring back generate-in-docker make target
On mac OSX the `make generate` also produces an extra files prefixed with 0, e.g. `manifests/setup/prometheus-operator-0prometheusruleCustomResourceDefinition`. This works fine when built within the container, but the problem is my env differs from that, so I need the docker generate for reproducibility. Unless there is a workaround for this, but I haven't found any so far apart from every time removing that file which is tedious.

cc @paulfantom @dgrisonnet 

cc @kakkoyun did you find any solutions for this?

### LINKED ISSUE #515: Make generate create two files without yaml extension in json format
## steps
- git clone current project
- make generate

## Result:
- two additional files without .yaml extension are created in json format:
prometheus-operator-0servicemonitorCustomResourceDefinition
prometheus-operator-0prometheusruleCustomResourceDefinition

Snip20200429_3

Attach these two files
[manifests.zip](https://github.com/coreos/kube-prometheus/files/4550560/manifests.zip)

### PR DESCRIPTION
**Pull request by dgrisonnet**
Fix #498 
Fix #515

### CODE REVIEW

**Comment by dgrisonnet** [comment]
@lilic @kakkoyun would you mind telling me if this does the trick?

Since we are also generating a `kustomization` json file outside of the manifests directory, I wanted to try this approach instead of removing the file manually.

**Comment by lilic** [comment]
Let me verify, missed this somehow!

File: build.sh [hunk 1]
```diff
@@ -15,5 +15,5 @@ rm -rf manifests
 mkdir -p manifests/setup
 
                                                # optional, but we would like to generate yaml, not json
```
Comment (lilic) [inline build.sh:17]: Nit: Can you fix the indent here, thanks! :)

**Comment by lilic** [comment]
Sadly this errors out for me.

> find: -delete: manifests/setup/0namespace-namespace: relative path potentially not safe
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: {}: No such file or directory
find: -delete: manifests/setup/prometheus-operator-clusterRole: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-clusterRoleBinding: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-deployment: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-service: relative path potentially not safe
find: -delete: manifests/setup/prometheus-operator-serviceAccount: relative path potentially not safe
make: *** [manifests] Error 1

File: README.md [hunk 2]
```diff
@@ -240,7 +240,7 @@ rm -rf manifests
 mkdir -p manifests/setup
 
                                                # optional, but we would like to generate yaml, not json
-jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; rm -f {}' -- {}
+jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; find {} -delete' -- {}
```
Comment (lilic) [inline README.md:243]: The solution that @kakkoyun suggested https://github.com/coreos/kube-prometheus/issues/515#issuecomment-621105820
 works for me 👍 But I think it needs to be verified by someone from linux side as well :D 

> -jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml; find {} -delete' -- {}
+jsonnet -J vendor -m manifests "${1-example.jsonnet}" | xargs -I{} sh -c 'cat {} | gojsontoyaml > {}.yaml' -- {}
+find manifests -type f ! -name '*.yaml' -delete

Comment (brancz) [reply README.md:243]: we already do this in other places, so I'm happy with this solution and it's tested on linux

File: build.sh (hunk 1 above)
Comment (dgrisonnet) [reply build.sh:17]: I believe this was meant to apply to the call to `gojsontoyaml`, I have updated the indentation and the comment to reflect that.

**Comment by dgrisonnet** [comment]
@brancz @lilic I addressed your comments, PTAL

**Comment by brancz** [comment]
Awesome, thank you!

**Pull request merged**